batch7crud-roll4/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,215 @@
+        view.forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String firstName = request.getParameter("firstName");
+        String lastName = request.getParameter("lastName");
+        String station = request.getParameter("station");
+
+        Employee employee = new Employee();
+        employee.setFirstName(firstName);
+        employee.setLastName(lastName);
+        employee.setStation(station);
+
+        try {
+            employeeService.save(employee);
+            response.sendRedirect(request.getContextPath() + "/employees");
+        } catch (DataException e) {
+            e.printStackTrace();
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+
+            RequestDispatcher view = request.getRequestDispatcher("/WEB-INF/views/error.jsp");
+            view.forward(request, response);
+        }
+    }
+
+    private void edit(HttpServletRequest request, HttpServletResponse response, int id) throws ServletException, IOException {
+        try {
+            String forward = "/WEB-INF/views/edit.jsp";
+            request.setAttribute("employee", employeeService.fetchById(id));
+
+            RequestDispatcher view = request.getRequestDispatcher(forward);
+            view.forward(request, response);
+        } catch (DataException e) {
+            e.printStackTrace();
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+
+            RequestDispatcher view = request.getRequestDispatcher("/WEB-INF/views/error.jsp");
+            view.forward(request, response);
+        }
+
+    }
+
+    private void editProcess(HttpServletRequest request, HttpServletResponse response, int id) throws ServletException, IOException {
+        String firstName = request.getParameter("firstName");
+        String lastName = request.getParameter("lastName");
+        String station = request.getParameter("station");
+
+        Employee employee = new Employee();
+        employee.setId(id);
+        employee.setFirstName(firstName);
+        employee.setLastName(lastName);
+        employee.setStation(station);
+
+        try {
+            employeeService.update(employee);
+            response.sendRedirect(request.getContextPath() + "/employees");
+        } catch (DataException e) {
+            e.printStackTrace();
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+
+            RequestDispatcher view = request.getRequestDispatcher("/WEB-INF/views/error.jsp");
+            view.forward(request, response);
+        }
+
+    }
+
+    private void deleteProcess(HttpServletRequest request, HttpServletResponse response, int id) throws ServletException, IOException {
+        try {
+            employeeService.deleteEmployee(id);
+            response.sendRedirect(request.getContextPath() + "/employees");
+        } catch (DataException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)